sympy/polys/domains/mpelements.py

```diff
@@ -22,10 +22,7 @@ def _set_mpf(self, val):
         prec, rounding = self.context._prec_rounding
         tol = self.context.tol
 
-        if mpf_lt(mpf_abs(val, prec, rounding), tol):
-            self.__mpf__ = fzero
-        else:
-            self.__mpf__ = val
+        self.__mpf__ = val
```

Comment:
Note: this particular truncation code was introduced in https://github.com/sympy/sympy/commit/d191d9379f45ca472b018c928a0e3954ee6efee3#diff-5cab79e07eace0f7c37ec2a8d2f131c6R19 as part of #2126.